pkg/cli/proxy_get.go

```diff
@@ -65,9 +65,11 @@ func GetEnvoyProxyConfig(clientSet kubernetes.Interface, config *rest.Config, na
 	return envoyProxyConfig, nil
 }
 
-// proxyLabelExists returns a boolean indicating if the pod is part of a mesh
+// proxyLabelExists returns a boolean indicating if the pod is part of a mesh.
+// Note that this is NOT a definitive check indicating pod membership in a mesh
+// as the label may continue to exist even if the pod is removed from the mesh.
 func proxyLabelExists(pod corev1.Pod) bool {
-	// osm-controller adds a unique label to each pod that belongs to a mesh
+	// osm-controller adds a unique label to a pod when it is added to a mesh
```

Comment:
Addresses @SanyaKochhar 's comments in https://github.com/openservicemesh/osm/pull/3881#discussion_r680086117 and https://github.com/openservicemesh/osm/pull/3881#discussion_r680089278

Reply:
@johnsonshi could you expose this function for reuse in osm-health?